code/game/gamemodes/game_mode.dm

```diff
@@ -541,3 +546,19 @@ proc/get_nt_opposed()
 			to_chat(usr, "[ticker.mode.extended_round_description]")
 	else
 		to_chat(usr, "<i>Shhhh</i>. It's a secret.")
+
+//Handlers for people dying / being revived.
+/datum/game_mode/proc/on_crew_death(mob/living/carbon/human/H)
+	if(!H)
+		return
+	player_count = GLOB.all_crew_records.len
+	dead_players += H
+	check_finished()
```

Comment:
check finished should just be in the marker gamemode's override of this function